### DISCUSSION TITLE
supervision-0.13.0

### DISCUSSION THREAD

**Discussion by SkalskiP** [roboflow/supervision#276] [Announcements]
## 🚀 Added

- Support for mean average precision (mAP) for object detection models with [`sv.MeanAveragePrecision`](https://roboflow.github.io/supervision/metrics/detection/#meanaverageprecision). ([#236](https://github.com/roboflow/supervision/pull/236))

```python
>>> import supervision as sv
>>> from ultralytics import YOLO

>>> dataset = sv.DetectionDataset.from_yolo(...)

>>> model = YOLO(...)
>>> def callback(image: np.ndarray) -> sv.Detections:
...     result = model(image)[0]
...     return sv.Detections.from_yolov8(result)

>>> mean_average_precision = sv.MeanAveragePrecision.benchmark(
...     dataset = dataset,
...     callback = callback
... )

>>> mean_average_precision.map50_95
0.433
```

- Support for `ByteTrack` for object tracking with [`sv.ByteTrack`](https://roboflow.github.io/supervision/tracker/core/#bytetrack). ([#256](https://github.com/roboflow/supervision/pull/256))

```python
>>> import supervision as sv
>>> from ultralytics import YOLO

>>> model = YOLO(...)
>>> byte_tracker = sv.ByteTrack()
>>> annotator = sv.BoxAnnotator()

>>> def callback(frame: np.ndarray, index: int) -> np.ndarray:
...     results = model(frame)[0]
...     detections = sv.Detections.from_yolov8(results)
...     detections = byte_tracker.update_from_detections(detections=detections)
...     labels = [
...         f"#{tracker_id} {model.model.names[class_id]} {confidence:0.2f}"
...         for _, _, confidence, class_id, tracker_id
...         in detections
...     ]
...     return annotator.annotate(scene=frame.copy(), detections=detections, labels=labels)

>>> sv.process_video(
...     source_path='...',
...     target_path='...',
...     callback=callback
... )
```

https://github.com/roboflow/supervision/assets/26109316/d5d393f5-e577-474a-bc8c-82483ef8a578

- [`sv.Detections.from_ultralytics`](https://roboflow.github.io/supervision/detection/core/#supervision.detection.core.Detections.from_ultralytics) to enable seamless integration with [Ultralytics](https://github.com/ultralytics/ultralytics) framework. This will enable you to use `supervision` with all [models](https://docs.ultralytics.com/models/) that Ultralytics supports. ([#222](https://github.com/roboflow/supervision/pull/222))

    > **Warning**
    > [`sv.Detections.from_yolov8`](https://roboflow.github.io/supervision/detection/core/#supervision.detection.core.Detections.from_yolov8) is now deprecated and will be removed with `supervision-0.15.0` release.

- [`sv.Detections.from_paddledet`](https://roboflow.github.io/supervision/detection/core/#supervision.detection.core.Detections.from_paddledet) to enable seamless integration with [PaddleDetection](https://github.com/PaddlePaddle/PaddleDetection) framework. ([#191](https://github.com/roboflow/supervision/pull/191))

- Support for loading PASCAL VOC segmentation datasets with [`sv.DetectionDataset.`](https://roboflow.github.io/supervision/dataset/core/#supervision.dataset.core.DetectionDataset.from_pascal_voc). ([#245](https://github.com/roboflow/supervision/pull/245))

## 🏆 Contributors

@hardikdava (Hardik Dava), @kirilllzaitsev (Kirill Zaitsev), @onuralpszr (Onuralp SEZER), @dbroboflow (Dervis Basbabasi), @mayankagarwals (Mayank Agarwal), @danigarciaoca (Daniel M. García-Ocaña), @capjamesg (James Gallagher), @SkalskiP (Piotr Skalski)

<hr /><em>This discussion was created from the release <a href='https://github.com/roboflow/supervision/releases/tag/0.13.0'>supervision-0.13.0</a>.</em>

**Comment by maddust** [comment]
@SkalskiP there is a typo in the documentation   detections = byte_tracker.update_ from_detections(detections=detections)   ------->   detections = byte_tracker.update_with _detections(detections=detections) 

BTW great work guys. you have been killing it !

**Reply by SkalskiP** [reply]
Hi, @maddust 👋🏻! Thanks for pointing that out. 🙏🏻 I just updated the docs: https://roboflow.github.io/supervision/tracker/core/#supervision.tracker.byte_tracker.core.ByteTrack.update_with_detections

**Comment by epochDVKHN** [comment]
@SkalskiP is the redesigned annotators with the trail annotations out in this release? 

Yeah I agree, supervision is simply amazing 🤩

**Reply by hardikdava** [reply]
Hi @ElNoSabe322 :wave, thanks for taking an interest in the feature. We are already working on redesigning annotators. The feature will be available in next releases.

**Reply by SkalskiP** [reply]
Hi, @ElNoSabe322 👋🏻! As @hardikdava said, the work on new annotators is in the advanced stage, but we were not ready for `supervision-0.13.0`. I hope we will be ready to release it with `supervision-0.14.0`. Stay tuned.

**Comment by anshulsdeshpande** [comment]
I am getting followin error. Can anyone please help:
IndexError: boolean index did not match indexed array along dimension 0; dimension is 93 but corresponding boolean dimension is 94

**Reply by LinasKo** [reply]
Hi @anshulsdeshpande :wave:

It seems like an array access is incorrect somewhere. Can you please share the code you're running?

**Reply by anshulsdeshpande** [reply]
Hi @LinasKo 
I am running following code for object detection and counting. however, I am running into the mentioned error.
https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-track-and-count-vehicles-with-yolov8.ipynb
![image](https://github.com/roboflow/supervision/assets/85047984/faf92884-36f2-4852-a22f-b69ee6ec5b2e)

**Reply by LinasKo** [reply]
I see, the notebook uses a very old supervision version.

I'll see if I can make it compatible with the more recent developments.

**Reply by anshulsdeshpande** [reply]
@LinasKo updated supervision ( version =0.21.0rc3) worked. Thanks a lot :)

**Reply by SkalskiP** [reply]
@anshulsdeshpande also here is updated version of that notebook: https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-track-and-count-vehicles-with-yolov8-and-supervison.ipynb

**Reply by LinasKo** [reply]
@anshulsdeshpande, one final remark: I think it'd be best to use `supervision`.
That'll guarantee you future updates. I initially assumed that one was failing  :slightly_smiling_face:
